kube-runtime/src/reflector/dispatcher.rs

```diff
@@ -0,0 +1,361 @@
+use core::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use std::{fmt::Debug, sync::Arc};
+
+use derivative::Derivative;
+use futures::Stream;
+use pin_project::pin_project;
+use std::task::ready;
+
+use crate::reflector::{ObjectRef, Store};
+use async_broadcast::{InactiveReceiver, Receiver, Sender};
+
+use super::Lookup;
+
+#[derive(Derivative)]
+#[derivative(Debug(bound = "K: Debug, K::DynamicType: Debug"), Clone)]
+// A helper type that holds a broadcast transmitter and a broadcast receiver,
+// used to fan-out events from a root stream to multiple listeners.
+pub(crate) struct Dispatcher<K>
+where
+    K: Lookup + Clone + 'static,
+    K::DynamicType: Eq + std::hash::Hash + Clone,
+{
+    dispatch_tx: Sender<ObjectRef<K>>,
+    // An inactive reader that prevents the channel from closing until the
+    // writer is dropped.
+    _dispatch_rx: InactiveReceiver<ObjectRef<K>>,
+}
+
+impl<K> Dispatcher<K>
+where
+    K: Lookup + Clone + 'static,
+    K::DynamicType: Eq + std::hash::Hash + Clone,
+{
+    pub(crate) fn new(buf_size: usize) -> Dispatcher<K> {
```

Comment:
This public ctor probably should have a comment about the buf_size and how it relates to backpressure